DataFormats/L1TrackTrigger/interface/TTCluster.h

```diff
@@ -0,0 +1,185 @@
+/*! \class   TTCluster
+ *  \brief   Class to store the L1 Track Trigger clusters
+ *  \details After moving from SimDataFormats to DataFormats,
+ *           the template structure of the class was maintained
+ *           in order to accomodate any types other than Phase2TrackerDigis
+ *           in case there is such a need in the future.
+ *
+ *  \author Nicola Pozzobon
+ *  \author Emmanuele Salvati
+ *  \date   2013, Jul 12
+ *
+ */
+
+#ifndef L1_TRACK_TRIGGER_CLUSTER_FORMAT_H
+#define L1_TRACK_TRIGGER_CLUSTER_FORMAT_H
+
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/Common/interface/Ptr.h"
+#include "DataFormats/Common/interface/DetSet.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/Phase2TrackerDigi/interface/Phase2TrackerDigi.h"
+#include "DataFormats/GeometryCommonDetAlgo/interface/MeasurementPoint.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h" /// NOTE: this is needed even if it seems not
+
+template< typename T >
+class TTCluster
+{
+  public:
+    /// Constructors
+    TTCluster();
+    TTCluster( std::vector< T > aHits,
+               DetId aDetId,
+               unsigned int aStackMember,
+               bool storeLocal );
+
+    /// Destructor
+    ~TTCluster();
+
+    /// Data members:   getABC( ... )
+    /// Helper methods: findABC( ... )
+
+    /// Hits in the Cluster
+    std::vector< T > getHits() const                   { return theHits; }
+    void             setHits( std::vector< T > aHits ) { theHits = aHits; }
+
+    /// Detector element
+    DetId        getDetId() const         { return theDetId; }
+    void         setDetId( DetId aDetId ) { theDetId = aDetId; }
+    unsigned int getStackMember() const                      { return theStackMember; }
+    void         setStackMember( unsigned int aStackMember ) { theStackMember = aStackMember; }
+
+    /// Rows and columns to get rid of Digi collection
+    std::vector< int > findRows() const;
+    std::vector< int > findCols() const;
+    void setCoordinates( std::vector< int > a, std::vector< int > b ) { theRows = a; theCols = b; }
+    std::vector< int > getRows() const { return theRows; }
+    std::vector< int > getCols() const { return theCols; }
+
+    /// Cluster width
+    unsigned int findWidth() const;
+
+    /// Single hit coordinates
+    /// Average cluster coordinates
+    MeasurementPoint findHitLocalCoordinates( unsigned int hitIdx ) const;
+    MeasurementPoint findAverageLocalCoordinates() const;
+
+    /// Information
+    std::string print( unsigned int i = 0 ) const;
+
+  private:
+    /// Data members
+    std::vector< T >  theHits;
```

Comment:
Unfortunately not. In general the size is small but it can get very large in some cases. The only formal maximum is the number of strips in the module.